### PR TITLE
#12337 Release 24.10.0 

### LINKED ISSUE #12337: Release 24.10.0 with support for Python 3.13
This is blocking on https://github.com/twisted/twisted/pull/12336

The idea is to have a new release with support for Python 3.13.0

### PR DESCRIPTION
**Pull request by adiroiban**
## Scope and purpose

Fixes #12337

This is the release for 24.10.0

process documented at https://docs.twisted.org/en/stable/development/release-process.html

### CODE REVIEW

**Comment by adiroiban** [comment]
GitHub release at https://github.com/twisted/twisted/releases/tag/twisted-24.10.0rc1
PyPi release at https://pypi.org/project/Twisted/24.10.0rc1/

This is ready for review

needs-review

File: NEWS.rst [hunk 1]
```diff
@@ -8,6 +8,113 @@ https://twisted.org/trac/ticket/<number>
...
+
+- twisted.conch.ssh.keys.Key can now load public blob keys of type sk-ssh-ed25519@openssh.com and sk-ecdsa-sha2-nistp256@openssh.com. (#12211)
+
+
+Bugfixes
+~~~~~~~~
+
+- twisted.conch tests no longer rely on OpenSSH supporting DSA keys, fixing compatibility with OpenSSH >= 9.8. (#12273)
+- twisted.conch.ssh.SSHCiphers no longer supports the cast128-ctr, cast128-cbc, blowfish-ctr, and blowfish-cbc ciphers.
+  The Blowfish and CAST5 ciphers were removed as they were deprecated by the Python cryptography library.``` (#12308)
```
Comment (twm) [inline NEWS.rst]: Some misplaced backticks here?

```suggestion
  The Blowfish and CAST5 ciphers were removed as they were deprecated by the Python cryptography library. (#12308)
```

File: NEWS.rst [hunk 2]
```diff
@@ -8,6 +8,113 @@ https://twisted.org/trac/ticket/<number>
...
+
+- Python 3.13 is now supported. (#11750)
+- twisted.internet.defer.succeed() is significantly faster, and awaiting Deferred has also been sped up. (#12225)
+- twisted.python.failure.Failure creation no longer records the place where it was created. This reduces creation time by 60% at least, thereby speeding up Deferred error handling. (#12230)
+- twisted.internet.defer.Deferred no longer removes the traceback object from Failures. This may result in more objects staying in memory if you don't clean up failed Deferreds, but it speeds up error handling and enables improvements to traceback reporting. (#12234)
+- twisted.internet.defer APIs are 2%-4% faster in many cases. (#12237)
+- twisted.internet.defer.Deferred runs callbacks with chained Deferreds a little faster. (#12284)
+- The reactor now will use a little less CPU when events have been scheduled with callLater(). (#12286)
+- Creation of twisted.python.failure.Failure is now faster. (#12288)
+- twisted.web.server is 1-2% faster in some cases. (#12318)
```
Comment (twm) [inline NEWS.rst]: Shouldn't this be down in the web section?

**Review by twm — APPROVED** [review]
Just minor notes, looks good. Thank you for managing the release @adiroiban!

File: NEWS.rst (hunk 2 above)
Comment (adiroiban) [reply NEWS.rst]: True.

I will update this before the release.

---------

I would prefer to have just a single newsfragment directory for the whole project.

In recent years, twisted is mostly common async + HTTP + SSH

Many releases are done without changes to `words`, `mail` or `names` ... and `conch` and `trial` usually get one or two changes per release.

**Comment by adiroiban** [comment]
Thanks again for the review

The final release was triggered via the tag and the publish process is here https://github.com/twisted/twisted/actions/runs/11545295320

pypi ready at https://pypi.org/project/Twisted/24.10.0/

I will update to dev version and will merge

**Pull request merged**